packages/js/product-editor/src/blocks/downloads/edit.tsx

```diff
@@ -0,0 +1,243 @@
+					key: String( download.id ),
+					download,
+				} );
+				return items;
+			}, fileItemsInDownloads );
+		} );
+	}, [ downloads ] );
+
+	function handleFileUpload( files: MediaItem[] ) {
+		const { uploadedFiles, items } = files.reduce< {
+			uploadedFiles: Product[ 'downloads' ];
+			items: DownloadableFileItem[];
+		} >(
+			( current, file, index ) => {
+				const download = {
+					id: file.id ? String( file.id ) : '',
+					file: file.url,
+					name: getFileName( file.url ),
+				};
+				const item = {
+					key: `_${ index }`,
+					download,
+					uploading: ! Boolean( file.id ),
+				};
+
+				if ( download.id ) {
+					current.uploadedFiles.push( download );
+				}
+				current.items.push( item );
+
+				return current;
+			},
+			{ uploadedFiles: [], items: [] }
+		);
+
+		if ( uploadedFiles.length ) {
+			if ( ! downloads.length ) {
+				setDownloadable( true );
+			}
+			setDownloads( [ ...downloads, ...uploadedFiles ] );
+		}
+
+		setFileItems( items );
+	}
+
+	function removeHandler( fileItem: DownloadableFileItem ) {
+		return function handleRemoveClick() {
+			const otherDownloads = downloads.reduce< ProductDownload[] >(
+				function removeDownload( others, current ) {
+					if (
+						String( current.id ) === String( fileItem.download.id )
+					) {
+						return others;
+					}
+					return [ ...others, current ];
+				},
+				[]
+			);
+
+			if ( ! otherDownloads.length ) {
+				setDownloadable( false );
+			}
+
+			setDownloads( otherDownloads );
+		};
+	}
+
+	return (
+		<div { ...blockProps }>
+			{ Boolean( fileItems.length ) ? (
+				<Sortable className="wp-block-woocommerce-product-downloads-field__table">
+					{ fileItems.map( ( fileItem ) => {
+						const nameFromUrl = getFileName(
+							fileItem.download.file
+						);
+
+						return (
+							<ListItem key={ String( fileItem.key ) }>
+								<div className="wp-block-woocommerce-product-downloads-field__table-filename">
+									<span>{ fileItem.download.name }</span>
+									{ fileItem.download.name !==
+										nameFromUrl && (
+										<span className="wp-block-woocommerce-product-downloads-field__table-filename-description">
+											{ nameFromUrl }
+										</span>
+									) }
+								</div>
+
+								<div className="wp-block-woocommerce-product-downloads-field__table-actions">
+									{ fileItem.uploading && (
+										<Spinner
+											aria-label={ __(
+												'Uploading file',
+												'woocommerce'
+											) }
+										/>
+									) }
+									<Button
+										icon={ closeSmall }
+										label={ __(
+											'Remove file',
+											'woocommerce'
+										) }
+										disabled={ fileItem.uploading }
+										onClick={ removeHandler( fileItem ) }
+									/>
+								</div>
+							</ListItem>
+						);
+					} ) }
+				</Sortable>
+			) : (
+				<MediaUploader
+					label={
+						<>
+							<UploadImage />
+							<p className="woocommerce-product-form__remove-files-drop-zone-label">
```

Comment:
Can we rename this className to `woocommerce-product-form__supported-files-drop-zone-label`?

Reply:
done -> https://github.com/woocommerce/woocommerce/pull/40526/commits/82381641356554dabd2c8794593f35ec28857997